Game/world/npc.cpp

```diff
@@ -3088,6 +3088,10 @@ SensesBit Npc::canSenseNpc(float tx, float ty, float tz, bool freeLos, float ext
   if(!freeLos){
     float dx  = x-tx, dz=z-tz;
     float dir = angleDir(dx,dz);
+    if((bodyState()&BodyState::BS_SIT)!=0 &&    //If we are sitting with
+       (visual.pose().isInAnim("S_BENCH_S1") || //this ani on a bench or
```

Comment:
I don't think hard-coding animation for this case is a correct approach.
As mention in issue:
> Seem like Npc_CanSeeNpc should use BIP01 rotation, to calculate eye-dir vector

Reply:
I don't know what BIP01 rotation is.
I'm not that familar with the animation system in detail, yet.
Is it hard to do ?

I know, this here is just a quick fix to get it working as fast as possible ;-)
Shame on me.

Reply:
In Gothic, every bone has a name, "BIP01" is one of them.

It's possible to query a bone index by calling `Skeleton::findNode("BIP01")`
Once you havea `id`, it's possible to fetch a transformed bone matrix:
`Matrix4x4 tr = skInst->tr[id]`
Most-likely this matrix suppose to be used, to deduce a 'real' rotation of npc.

You can find more details about bone-usage in mdlvisual.cpp

Reply:
Ok, if I broke it down correctly, here is my proposal how to get the head rotation:
npc [Npc] ->visual [MdlVisual]->head [Mesh] -> protoMesh() -> Node.transform
- Extract rotations from ObjectMatrix to get an atVector (sub Translation, unapply scale, build rotation matrix, convert to quaternion, get rotations as vector).
If I understand the Bone animation things correctly, the ObjectMatrix of the Head Mesh should include all transformations from the Base meshes, or ? Otherwise this might have to be done for all Base meshes, too, to get the final rotation of the head.

Downside of this approach, even if this is more correct, it's expensive and might introduce "bugs" during gameplay (npcs field of view changes). NPCs that might not have seen the hero before, can now see him related to the animations of the character. But, it might not be a problem.

What do you thing about that ?

Reply:
But I guess, as we would have to change the code to get the angle in CanSenseNpc anyway, the temporary workaround with the animations wouldn't be forgotten to be removed later.
So as a temporary workaround, it could do the job, maybe flagged with a fixme comment to replace it with the real head rotation, if implemented.

Reply:
You don't have to do so much work to fetch the transform, because they already calculated:
`Pose::tr' is a ready-to use matrices, in object space. Then to get a view vector:
```c++
  auto p = pos;
  if(boneId<pose.tr.size())
    p.mul(pose.tr[boneId]);
  Vec3 viewDir = Vec3(p.at(0,0),p.at(0,1),p.at(0,2));
``` 
Note: I don't know for sure which axis is used as a 'forward', but most-likely it's X-axis.

Not about possible bugs: idea is to pick a 'root' bone, aka BIP01, to use instead of 'BIP01 HEAD'. That way implementation can be more reliable.